imagebuildah/build.go

```diff
@@ -1538,8 +1542,22 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
 			cleanupStages[stage.Position] = stageExecutor
 		}
 
+		// ForceCommit the stage if this is the finalStage
+		// or if we need to use this stage's image as
+		// a base image later
+		forceCommit := stageIndex == len(stages)-1
+		if !forceCommit {
+			for index, value := range b.stages {
+				if value == stageExecutor {
+					continue
+				}
+				if index != stage.Name {
+					forceCommit = true
+				}
```

Comment:
I think that when we have multiple stages, this loop ends up always setting `forceCommit` to `true`?